vignettes/view-composition.Rmd

```diff
@@ -0,0 +1,186 @@
+---
+title: "View Composition"
+author: "Alicia Schep"
+date: "`r Sys.Date()`"
+output: rmarkdown::html_document
+editor_options: 
+  chunk_output_type: inline
+---
+
+```{r setup, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+## Combining multiple data displays
+
+Vegalite and Altair have several methods to help combine different views of these data.  
+
+* [**Layering**](#layer): Combine several displays on the same set of axes
+* [**Horizontal Concatenation**](#hconcat): Position several displays next to each other, horizontally.
+* [**Vertical Concatenation**](#vconcat): Position several displays next to each other, vertically
+* [**Faceting**](#facet): Create panels showing the same view of different subsets of the data
+* [**Repeating**](#repeat): Create panels that differ in the encoding of one or more channels
+
+For the first three methods, Altair has functions that take in two chart objects. Those chart objects can either be a simple chart made by `alt$Chart`, or a compound chart made by any of the methods above.  Altair and this R package also provides operators for these methods:
+
+| Operation  | Function Syntax | Operator Syntax |
+| ------------- | ------------- | ------------- |
+| Layering  | `alt$layer(chart1, chart2)`  | `chart1 + chart2`  |
+| Horizontal Concatenation  | `alt$hconcat(chart1, chart2)`  | `chart1 | chart2`  |
+| Vertical Concatenation  | `alt$vconcat(chart1, chart2)`  | `chart1 & chart2`  |
+
+In contrast, faceting and repeating are methods for Chart objects, so if you had created a chart called `chart1` you would call them using `chart1$`:
+
+| Operation  | Function Syntax |
+| ------------- | ------------- |
+| Faceting | `chart1$facet(...)`  |
+| Repeating  | ``chart1$`repeat`(...)``  |
+
+Note that for "repeat", backticks are required around the word `repeat`. This is because `repeat` is a [reserved word](http://stat.ethz.ch/R-manual/R-devel/library/base/html/Reserved.html) in R!
+
+## Layered Charts {#layer}
+
+Layering involves combining multiple displays on the same set of axes. Two chart objections can be layered using either the `+` operator or the `alt$layer` function. A common use case is to add more than one mark for the same set of encodings.  For example, you could combine points and a line for time-series data. Here is an example using the "beaver1" data, which has body temparature for a beaver over time:
+
+```{r}
+library(altair)
+base = alt$Chart(r_to_py(beaver1))$encode(
+    x = alt$X('time'),
+    y = alt$Y('temp', scale = alt$Scale(zero = FALSE))
+)
+
+scatter_plot <- base$mark_point()
+line_plot <- base$mark_line()
+
+vegalite(scatter_plot + line_plot, height = 400)
```

Comment:
I don't know where is the right place to discuss, but I think the issue of [repeating the data](https://altair-viz.github.io/user_guide/compound_charts.html#advanced-compound-charts-and-data-specification) merits attention.

In the future, I think there is an opportunity for us on the R side (I will write this up as a separate issue) to provide some helper functions. I think we could use some **purrr**  magic to provide functions like:

- `remove_data()`, which would go through a chart and, well, remove the data.
- `distill_data()`, which would go through a compound chart, see if there are repeated data, and if so remove them in favor of a `data` at the "top" of the object.

Reply:
Yes, I was thinking that would be a good thing to add to the vignette eventually.  I like the idea of some helper functions!   It might even make sense for vegalite function to try to do some of that automatically (perhaps with an argument that enables turning that off if not desired).  Along similar lines, and `add_data` function might also be a nice helper function for adding data at the top level.